File: sw_x/gvm_core/internal/protorpc/csharp/ProtoChannel.cs
//
//  Copyright 2011 iGware Inc.
//  All Rights Reserved.
//
//  THIS SOFTWARE CONTAINS CONFIDENTIAL INFORMATION AND
//  TRADE SECRETS OF IGWARE INC.
//  USE, DISCLOSURE OR REPRODUCTION IS PROHIBITED WITHOUT
//  THE PRIOR EXPRESS WRITTEN PERMISSION OF IGWARE INC.
//
using System;

namespace protorpc
{
    abstract public class ProtoChannel
    {
        /**
         * Serializes and writes the message to the channel.
         */
        abstract public void writeMessage(object message);
        
        /**
         * Reads and deserializes the message from the channel.
         */
        abstract public void extractMessage(Type type, ref object message);
        
        /** Flushes the output stream. */
        abstract public bool flushOutputStream();
    }
}

File: sw_x/gvm_core/internal/protorpc/java/igware/protobuf/ProtoChannel.java
package igware.protobuf;

import com.google.protobuf.MessageLite;

public interface ProtoChannel {
    
    /**
     * Serializes and writes the message to the channel.
     */
    public void writeMessage(MessageLite message) throws Exception;
    
    /**
     * Reads and deserializes the message from the channel.
     */
    public void extractMessage(MessageLite.Builder messageBuilder) throws Exception;
    
    /** Flushes the output stream. */
    public boolean flushOutputStream() throws Exception;
}
